src/Runner.Worker/JobRunner.cs

```diff
@@ -126,6 +126,11 @@ public async Task<TaskResult> RunAsync(AgentJobRequestMessage message, Cancellat
                 _runnerSettings = HostContext.GetService<IConfigurationStore>().GetSettings();
                 jobContext.SetRunnerContext("name", _runnerSettings.AgentName);
 
+                if (jobContext.Global.Variables.TryGetValue(WellKnownDistributedTaskVariables.RunnerEnvironment, out var runnerEnvironment))
```

Comment:
We don't have an allow list before for the `RUNNER_*` since before this change all `RUNNER_*` are set by the runner, while most `GITHUB_*` are coming from the service.